### PR TITLE
feat(web2): added WiFi channels and Country Code support

### PR DESCRIPTION
**Pull request by marcellorinaldo**
This PR:
- adds support for WiFi channels and country code
- fixes also some configuration update bugs

**Related Issue:** N/A.

**Description of the solution adopted:** N/A.

**Screenshots:** N/A.

**Any side note on the changes made:** N/A.

### CODE REVIEW

**Review by mattdibi — CHANGES_REQUESTED** [review]
As per our call/offline discussion:
- handle the active channels selection in the GUI. Currently the active channel is always set to 1, indipendently of the actual selected channel.
- ~handle radio mode update so that the correctly supported channel can be selected from the dropdown menu.~ <- Was already working. My bad.

Everything else looks great to me!

**Review by mattdibi — CHANGES_REQUESTED** [review]
After merge #4425 the signature of `WifiAccessPoint` changed. It doesn't report the supported channels as list of integer anymore. Rebase and update.

**Comment by marcellorinaldo** [comment]
@mattdibi Rebase done.

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java [hunk 1]
```diff
@@ -82,6 +90,68 @@ public Optional<NetworkInterfaceType> getNetInterfaceType(String ifName) {
...
+        }
+
+        WifiInterfaceStatus wifiInterfaceInfo = (WifiInterfaceStatus) netInterface.get();
+        List<WifiChannel> channels = wifiInterfaceInfo.getChannels();
+
+        List<GwtWifiChannelFrequency> gwtChannels = new ArrayList<>();
+
+        for (WifiChannel channel : channels) {
+            gwtChannels.add(new GwtWifiChannelFrequency(channel.getChannel(), channel.getFrequency()));
+        }
```
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java]: I expected to find here the channel filtering logic. As per our offline discussion we introduced #4425 to be able to filter channels in the UI as per our old algorithm:

```java
private boolean shouldBeDisabled(GwtWifiChannelFrequency channelFrequency) {
        boolean isNotValidCountryCode = this.countryCode.getValue().isEmpty()
                || this.countryCode.getValue().equals(WORLD_REGION_COUNTRY_CODE);
        return channelFrequency.isDisabled()
                || channelFrequency.isNoIrradiation() && !channelFrequency.isRadarDetection()
                || channelFrequency.isNoIrradiation() && channelFrequency.isRadarDetection() && isNotValidCountryCode;
    }
 ```

The `WifiChannel` data structure allows you to find this information.

Am I missing something? Is the filtering logic somewhere else?

**Comment by mattdibi** [comment]
During testing I encountered multiple errors related to vanishing DBus objects

```
com.google.gwt.user.server.rpc.UnexpectedException: Service method 'public abstract java.util.List org.eclipse.kura.web.shared.service.GwtNetworkService.findNetInterfaceConfigurations(boolean) throws org.eclipse.kura.web.shared.GwtKuraException' threw an unexpected exception: org.freedesktop.dbus.errors.UnknownMethod: No such interface ?org.freedesktop.DBus.Properties? on object at path /org/freedesktop/NetworkManager/IP4Config/9
```

will be hopefully addressed in #4422

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java (hunk 1 above)
Comment (marcellorinaldo) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java]: Forgot about that, adding it.

**Comment by mattdibi** [comment]
After setting the interface in "Station Mode" I can no longer set it back to "Access Point". See video below:

https://user-images.githubusercontent.com/22748355/222697268-e9f6f6af-4986-4c95-ab08-f57df1760d14.mov

**Review by mattdibi — APPROVED** [review]
LGTM! All issues have been addressed and everything works as expected. GJ

**Pull request merged**